compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -328,7 +332,9 @@ trait ConstraintHandling {
         || {
           constraint = c1
           val TypeBounds(lo, hi) = constraint.entry(param): @unchecked
-          isSub(lo, hi)
+          val isSat = isSub(lo, hi)
+          if isSat then tryReplace(lo, hi) // isSub may have introduced new constraints
```

Comment:
Ah I see what I did wrong. I reused the bounds as they were prior to `isSub` when rechecking for equal bounds, in an attempt to avoid requiring the entry :/